File: scripts/sha-1/sha-1.c
#include <openssl/sha.h>
#include <stdio.h>
#include <string.h>

void compute_sha1(const char *input) {
    SHA_CTX ctx;                          // Context for SHA-1
    unsigned char digest[SHA_DIGEST_LENGTH];

    SHA1_Init(&ctx);                      // Initialize SHA-1
    SHA1_Update(&ctx, input, strlen(input)); // Update with data
    SHA1_Final(digest, &ctx);             // Finalize hash

    printf("SHA-1 hash computed.\n");

    //print digest
    for (int i = 0; i < SHA_DIGEST_LENGTH; i++) {
        printf("%02x", digest[i]);
    }
    printf("\n");
}

int main() {
    const char *data = "test";
    compute_sha1(data);
    return 0;
}

File: scripts/sha-1/sha-1.java
import java.security.MessageDigest;

public class SHA1Test {
    public static void main(String[] args) throws Exception {
        String data = "test";

        // SHA-1 Example
        MessageDigest sha1 = MessageDigest.getInstance("SHA-1");
        byte[] sha1Hash = sha1.digest(data.getBytes());
        System.out.println("SHA-1 hash computed.");
    }
}
